packages/controllers/src/permissions/rpc-methods/requestPermissions.test.ts

```diff
@@ -0,0 +1,97 @@
+import { JsonRpcEngine } from 'json-rpc-engine';
+import { ethErrors } from 'eth-rpc-errors';
+import { requestPermissionsHandler } from './requestPermissions';
+
+describe('requestPermissions RPC method', () => {
+  it('returns the values of the object returned by requestPermissionsForOrigin', async () => {
+    const { implementation } = requestPermissionsHandler;
+    const mockRequestPermissionsForOrigin = jest
+      .fn()
+      .mockImplementationOnce(() => {
```

Comment:
Hm, surely making it `async` is enough, no?

I added the missing test case in: dbe980e

Reply:
If we forget to `await` it, the test might still succeed because the code in an async function still gets run synchronously when it gets called (at least in some situations). `async` ensures it will block if it has to, but it's still eager.

A delay would prove that it was `await` properly.

Not a blocker but I have definitely seen broken functions with passing unit tests because of this problem.

Reply:
Yeah, fair enough! e14519b